### PR TITLE
Assignment5(WIP)

### PR DESCRIPTION
**Pull request by AaronSauther**

### CODE REVIEW

File: Mailbox/Size.cs [hunk 1]
```diff
@@ -1,4 +1,20 @@
-﻿namespace Mailbox
+﻿using System;
+
+namespace Mailbox
 {
+    [Flags]
+    public enum Size
```
Comment (dsergio) [inline Mailbox/Size.cs]: should be named Sizes because it uses Flags

File: Mailbox/Mailbox.cs [hunk 2]
```diff
@@ -1,4 +1,26 @@
...
+    public class Mailbox
+    {
+        public Size Size { get; set; }
+        public (int x, int y) Location { get; set; }
+        public Person Owner { get; set; }
 
-}
+        public Mailbox(Size size, (int x, int y) location, Person owner)
+        {
+            Size = size;
```
Comment (dsergio) [inline Mailbox/Mailbox.cs]: check here to make sure size isn't set to Small | Medium or another invalid value?

File: Mailbox.Tests/PersonTests.cs [hunk 3]
```diff
@@ -0,0 +1,122 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+
+namespace Mailbox.Tests
```
Comment (dsergio) [inline Mailbox.Tests/PersonTests.cs]: looks good, good naming conventions and test coverage

File: Mailbox.Tests/DataLoaderTests.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Mailbox.Tests
+{
+    [TestClass]
+    public class DataLoaderTests
```
Comment (dsergio) [inline Mailbox.Tests/DataLoaderTests.cs]: don't forget the DataLoaderTests

File: Mailbox/Program.cs [hunk 5]
```diff
@@ -89,17 +92,17 @@ static void Main(string[] args)
 
         public static string GetOwnersDisplay(Mailboxes mailboxes)
         {
-            
+            return null;
```
Comment (dsergio) [inline Mailbox/Program.cs]: don't forget to implement

File: Mailbox/Mailbox.cs (hunk 2 above)
Comment (AaronSauther) [reply Mailbox/Mailbox.cs]: That's a good point, thanks!

File: Mailbox.Tests/DataLoaderTests.cs [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    {
+        [TestMethod]
+        public void Load_OneMailbox_ReturnsCorrectValue()
+        {
+            //Arrange
+            Mailbox mailbox = new Mailbox(Sizes.Small, (0, 0), new Person("Homer", "Simpson"));
+            MemoryStream memoryStream = new MemoryStream();
+            using (StreamWriter streamWriter = new StreamWriter(memoryStream, new UTF8Encoding(), 1024, true))
```
Comment (Keboo) [inline Mailbox.Tests/DataLoaderTests.cs]: There are static helpers for getting access to the encodings.
```suggestion
            using (StreamWriter streamWriter = new StreamWriter(memoryStream, System.Text.Encoding.UTF8, 1024, true))
```

File: Mailbox/DataLoader.cs [hunk 7]
```diff
@@ -1,21 +1,83 @@
...
+            {
+                string mailboxString = streamReader.ReadLine() ?? "";
+                Mailbox mailbox = new Mailbox(Sizes.Default, (0, 0), new Person("", ""));
+                try
+                {
+                    mailbox = JsonConvert.DeserializeObject<Mailbox>(mailboxString);
+                }
+                catch (JsonReaderException jsonReaderException)
+                {
+                    Console.WriteLine(nameof(jsonReaderException));
```
Comment (Keboo) [inline Mailbox/DataLoader.cs]: Not sure this WriteLine adds much value here. Perhaps
```suggestion
                    Console.WriteLine(jsonReaderException.ToString());
```

File: Mailbox/DataLoader.cs [hunk 8]
```diff
@@ -1,21 +1,83 @@
...
+        private bool disposedValue = false;
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                }
+                Source.Dispose();
```
Comment (Keboo) [inline Mailbox/DataLoader.cs]: This should be put inside of the `if(disposing)`

File: Mailbox/Person.cs [hunk 9]
```diff
@@ -1,4 +1,52 @@
...
+        }
+
+        public static bool operator !=(Person left, Person right)
+        {
+            return !(left == right);
+        }
 
+        public override string ToString()
+        {
+            return $"{this.FirstName} {this.LastName}";
```
Comment (Keboo) [inline Mailbox/Person.cs]:
```suggestion
            return $"{FirstName} {LastName}";
```

File: Mailbox/Program.cs [hunk 10]
```diff
@@ -1,16 +1,19 @@
...
 
 namespace Mailbox
 {
-    class Program
+    internal class Program
     {
-        private const int Width = 50;
-        private const int Height = 10;
+        private const int Width = 2;
+        private const int Height = 1;
```
Comment (Keboo) [inline Mailbox/Program.cs]: I assume these were set for testing?

File: Mailbox/Program.cs [hunk 11]
```diff
@@ -89,17 +98,41 @@ static void Main(string[] args)
...
         {
-            
+            StringBuilder stringBuilder = new StringBuilder();
+            foreach (Mailbox mailbox in mailboxes)
+            {
+                stringBuilder.Append($"{mailbox.Owner.ToString()}, ");
+            }
+            if (stringBuilder.Length > 2)
+                stringBuilder.Remove(stringBuilder.Length - 2, 2);
+            return stringBuilder.ToString();
```
Comment (Keboo) [inline Mailbox/Program.cs]: The expectation was for the size to include " Premium" when that flag was set, however I realize that bullet point was really poorly worded.

**Pull request closed without merging**